src/components/DsfrPagination/DsfrPagination.vue

```diff
@@ -2,24 +2,73 @@
 export default {
   name: 'DsfrPagination',
   props: {
-    links: {
+    pages: {
       type: Array,
       required: true,
     },
     currentPage: {
       type: Number,
-      default: 1,
+      default: 0,
+    },
+    firstPageTitle: {
+      type: String,
+      default: 'Première page',
+    },
+    lastPageTitle: {
+      type: String,
+      default: 'Dernière page',
+    },
+    nextPageTitle: {
+      type: String,
+      default: 'Page suivante',
+    },
+    prevPageTitle: {
+      type: String,
+      default: 'Page précédente',
+    },
+    truncLimit: {
+      type: Number,
+      default: 5,
     },
   },
 
   emits: ['update:currentPage'],
-
+  computed: {
+    startIndex () {
+      return Math.min(this.pages.length - 1 - this.truncLimit, Math.max(this.currentPage - (this.truncLimit - this.truncLimit % 2) / 2, 0))
+    },
+    endIndex () {
+      return Math.min(this.pages.length - 1, this.startIndex + this.truncLimit)
+    },
+    displayedPages () {
+      return this.pages.length > this.truncLimit ? this.pages.slice(this.startIndex, this.endIndex + 1) : this.pages
+    },
+  },
   methods: {
-    getPreviousPage () {
-      return this.currentPage > 1 ? this.currentPage - 1 : 1
+    tofirstPage () {
+      this.toPage(0)
+    },
+    toPreviousPage () {
+      this.toPage(Math.max(0, this.currentPage - 1))
     },
-    getNextPage () {
-      return this.currentPage < this.links.length ? this.currentPage + 1 : this.links.length
+    toNextPage () {
+      this.toPage(Math.min(this.pages.length - 1, this.currentPage + 1))
+    },
+    toLastPage () {
+      this.toPage(this.pages.length - 1)
+    },
+    toPage (index) {
+      const page = this.pages[index]
+      // Redirection manuelle pour éviter que
+      // le href ne change avant que la redirection n'ait été faite
+      if (page.href) window.location.replace(page.href)
```

Comment:
Pourquoi faire cela alors qu'il y a un `@click.prevent` ?

Il faudrait utiliser `this.$router` et mettre des accolades :
```javascript
if (page.href) {
  this.$router.replace(page.href)
}
```

Reply:
Pour améliorer l'accessibilité je souhaitais mettre les href directement sur les liens (sans le prevent)
Malheureusement avec l'évol sur la troncature, au clic sur un lien, vue "va trop vite" : l'affichage se met à jour puis on est redirigé vers le href du nouveau lien.
J'ai donc empêché l'action par défaut pour gérer la redirection à la main. Du point de vue de l'utilisateur ça devrait être transparent.

Reply:
Est-ce que le style est correct même s’il n’y a pas de `href` ?

Reply:
Les liens ne sont plus cliquables s'il n'y a pas de href. Ils apparaissent disabled
Mais c'est normal, puisqu'on ne saurait pas vers quoi rediriger. 
Dans ce contexte, this.$router est undefined. Est-ce que j'ai mal configuré quelque chose ?

Reply:
Donc `page.href` sera toujours vrai, non ?

Reply:
Fonctionnellement oui, il y a peu de chances qu'on utilise ce composant avec des href vides.
Par contre techniquement le disabled n'est ajouté que grâce à une propriété CSS.

Mais on peut retirer le test si vous pensez que le cas est trop marginal

Reply:
Je ne souhaite pas donner au composant la responsabilité de changer de route. Le développeur qui utilisera le composant saura que l’utilisateur a demandé à changer de page avec l’événement `'update:currentPage'` et décidera lui-même de changer de route ou non.
Il faudrait donc enlever ces 3 lignes :
```
if (page.href) {
  this.$router.replace(page.href)
}
```

Reply:
J'ai modifié ma PR en ce sens

Reply:
Super, merci pour ta réactivité, et pardon pour la nôtre !
Tu veux bien la rebaser sur develop à jour, et on la fusionne dans la foulée ?

Reply:
Pas de problème !
C'est rebased